### ISSUE TITLE
SpectrogramPluginOptions are ignored?

### ISSUE THREAD

**Issue by joskezelensky** [katspaugh/wavesurfer.js#3952]
## Bug description

SpectrogramPluginOption(s) height doesn't/don't get looked at correctly? when i set height to x it just _defaults?_ to 64px
## Environment
 - Browser: Chrome

## Minimal code snippet
```js
// Spectrogram plugin
    import WaveSurfer from 'https://cdn.jsdelivr.net/npm/wavesurfer.js@7.8/dist/wavesurfer.esm.js'
    import Spectrogram from 'https://cdn.jsdelivr.net/npm/wavesurfer.js@7.8/dist/plugins/spectrogram.esm.js'
    // Create an instance of WaveSurfer
    const ws = WaveSurfer.create({
      container: '#waveform',
      waveColor: 'rgb(200, 0, 200)',
      progressColor: 'rgb(100, 0, 100)',
      url: 'audio.flac',
      sampleRate: 48000,
    })
    // Initialize the Spectrogram plugin
    ws.registerPlugin(
      Spectrogram.create({
        labels: true,
        height: 256,
        splitChannels: false,
      }),
    )
```
## Expected result

height to be correctly applied
## Obtained result

height gets set to 64px

**Comment by J0hn3ch** [comment]
> ## Bug description
> SpectrogramPluginOption(s) height doesn't/don't get looked at correctly? when i set height to x it just _defaults?_ to 64px
> 
> ## Environment
> * Browser: Chrome
> 
> ## Minimal code snippet
> ```js
> // Spectrogram plugin
>     import WaveSurfer from 'https://cdn.jsdelivr.net/npm/wavesurfer.js@7.8/dist/wavesurfer.esm.js'
>     import Spectrogram from 'https://cdn.jsdelivr.net/npm/wavesurfer.js@7.8/dist/plugins/spectrogram.esm.js'
>     // Create an instance of WaveSurfer
>     const ws = WaveSurfer.create({
>       container: '#waveform',
>       waveColor: 'rgb(200, 0, 200)',
>       progressColor: 'rgb(100, 0, 100)',
>       url: 'audio.flac',
>       sampleRate: 48000,
>     })
>     // Initialize the Spectrogram plugin
>     ws.registerPlugin(
>       Spectrogram.create({
>         labels: true,
>         height: 256,
>         splitChannels: false,
>       }),
>     )
> ```
> 
> ## Expected result
> height to be correctly applied
> 
> ## Obtained result
> height gets set to 64px
> 
> ## Screenshots

I have the same problem! 

![image](https://github.com/user-attachments/assets/b90c870c-72e1-41f4-a103-1f1438214091)

```
wavesurfer.registerPlugin(
        WaveSurfer.Spectrogram.create({
          labels: true,
          height: 256,
          splitChannels: true,
        }),
);
```
Specifications:
- Node.js + Express.js
- Embedded JavaScript Template Engine

**Comment by dzdydx** [comment]
Same issue here.
Actually, it still happens even if you change the `height` in [official example](https://wavesurfer.xyz/examples/?spectrogram.js). The height of spectrogram will always stay unchanged.

![image](https://github.com/user-attachments/assets/41f6931c-6c04-431f-8158-9c69f8c45377)

**Comment by dzdydx** [comment]
I think I know the reasons.

https://github.com/katspaugh/wavesurfer.js/blob/8368928ab2c4bcd307312aed057ba572849f56a2/src/plugins/spectrogram.ts#L351C1-L358C6
```
    this.numMelFilters = options.numMelFilters || this.fftSamples / 8
    if (this.scale == 'mel') {
        this.height = options.height || this.numMelFilters
        this.height = Math.min(this.height, this.numMelFilters)
    } else {
        this.height = options.height || this.fftSamples / 2
        this.height =  Math.min(this.height, this.fftSamples / 2)
    }
```
The default scale is mel scale, so the height of the spectrogram will be `Math.min(options.height, this.numMelFilters)`. Unfortunately, default `numMelFilters` is set to `this.fftSamples / 8`, where `this.fftSamples` is `512` by default.

Therefore, if one uses the default value of `fftSamples` and `numMelFilters`, then setting `height` to any value bigger than `64` will not work.

@katspaugh maybe I can try to fix this, in case you don't have enough time.

@J0hn3ch @joskezelensky To temporarily get over with this, you can set `fftSamples` to a higher value like `1024`.

**Comment by katspaugh** [comment]
@dzdydx thank you, makes sense! A fix would be awesome, you're very welcome to do that.

**Comment by J0hn3ch** [comment]
> Therefore, if one uses the default value of `fftSamples` and `numMelFilters`, then setting `height` to any value bigger than `64` will not work.

Thank you @dzdydx for the help! I want to make you notice also that in the documentation of the module plugin/spectrogram (here [https://wavesurfer.xyz/docs/types/plugins_spectrogram.SpectrogramPluginOptions](https://wavesurfer.xyz/docs/types/plugins_spectrogram.SpectrogramPluginOptions) ) there is no reference to `numMelFilters` in order to set it, and also in the code comments. So it's common that this is set to `fftSamples / 8` by default.

**Comment by dzdydx** [comment]
@J0hn3ch @joskezelensky guys, I think this issue is now fixed by #3796. Update to version 7.8.12 and everything should be OK.
